docs/pnp/migrate-vsto-to-office-web-add-in-shared-code-library-tutorial.md

```diff
@@ -0,0 +1,427 @@
+    > [!NOTE]
+    > Don't use the .NET Core class library because it will not work with your VSTO project.
+5. In the **Configure your new project** dialog, set the following fields.
+    - Set the  **Project name** to **CellAnalyzerSharedLibrary**.
+    - Leave the **Location** at it's default value.
+    - Set the **Framework** to **4.7.2**.
+6. Choose **Create**.
+7. After the project is created, rename the **Class1.cs** file to **CellOperations.cs**. You will be prompted to rename the class as well. Do that so the class name matches the file name.
+8. Add the following code to the `CellOperations` class to create a method named `GetUnicodeFromText`.
+
+```csharp
+public class CellOperations
+{
+    static public string GetUnicodeFromText(string value)
+    {
+        string result = "";
+        foreach (char c in value)
+        {
+            int unicode = c;
+
+            result += $"{c}: {unicode}\r\n";
+        }
+        return result;
+    }
+}
+```
+
+### Use the shared class library in the VSTO Add-in
+
+Now you need to update the VSTO Add-in to use the class library. This is important that both the VSTO Add-in and Office web add-in use the same shared class library so that future bug fixes or features are made in one location.
+
+1. In **Solution Explorer** expand the **Cell-Analyzer** project, right-click the **CellAnalyzerPane.cs** file, and choose **View Code**.
+2. In the `btnUnicode_Click` method, delete the following lines of code.
+
+```csharp
+//Convert to Unicode listing
+    string result = "";
+    foreach (char c in cellValue)
+    {
+        int unicode = c;
+        result += $"{c}: {unicode}\r\n";
+    }
+```
+
+3. Update the line of code under the `//Output the result` comment to read as follows:
+
+```csharp
+//Output the result
+txtResult.Text = CellAnalyzerSharedLibrary.CellOperations.GetUnicodeFromText(cellValue);
+```
+
+6. From the **Debug** menu, choose **Start Debugging**. The custom task pane should still work as expected. Enter some text in a cell, and then test that you can convert it to a Unicode list with the add-in.
+
+## Create a REST API wrapper
+
+The VSTO Add-in can use the shared class library directly since they are both .NET projects. However the Office web add-in won't be able to use .NET since it uses JavaScript. Next you will need to create a REST API wrapper. This enables the Office web add-in to call a REST API, which then passes the call along to the shared class library.
+
+1. If you haven't already, start Visual Studio 2019 and open the **\start\Cell-Analyzer.sln** solution.
+2. Right-click the solution in **Solution Explorer** and choose **Add > New Project**.
+3. In the **Add a new project dialog**, choose **ASP.NET Core Web Application** and choose **Next**.
+4. In the **Configure your new project** dialog, set the following fields.
+    - Set the  **Project name** to **CellAnalyzerRESTAPI**.
+    - Leave the **Location** at it's default value.
+5. Choose **Create**.
```

Comment:
Sidenote: I'm leaving all the "choose" commands because I think it's more "alternative-input friendly."  The menu commands should use "Select", so if you want to be a Microsoft Style stickler, you can change them. I.e. "On the **Blah** menu, select "blah." See[ https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui]( https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui)

Reply:
thanks!